Gray tweaks prototype

LeaVerou opened the pull request:
Core PR contributions:
- ✅ Pick gray undertone among core colors
- ✅ Tweak gray chroma (as "colorfulness"). 

**Issue:** currently this is using the same max colorfulness across all undertones (35% of the color) but it seems that some undertones (esp. the cooler ones, grays, indigos, purples) seem to be able to "take" more of it and still appear gray (e.g. the Awesome gray is at 34%!), whereas others (e.g. red) start looking like an actual hue scale. Should we have different scales per undertone?

Other things this PR does:
- Some refactor of palette tweaking code to make it more manageable
- Some tweaks on the UI showing that a palette is tweaked:
<img width="1037" alt="image" src="https://github.com/user-attachments/assets/f46535ac-ff3e-492f-8a71-fca902ff5e5e" />
- Added shadow to slider handle:
Before:
<img width="627" alt="image" src="https://github.com/user-attachments/assets/bb2d5195-8c6b-4aeb-acc0-4758a614e19e" />

After:
<img width="622" alt="image" src="https://github.com/user-attachments/assets/3f4980e4-9b59-4268-9684-4df2f517213f" />

Screencast of current behavior (note the tooltip bug is still there until #762 is merged):

https://github.com/user-attachments/assets/8f6f06f5-3652-4e23-86f6-9ca1a2ba1e8c

LeaVerou committed:
Save palette MVP

LeaVerou committed:
Iterate on UI

LeaVerou committed:
Fix

LeaVerou committed:
A bunch of fixes

LeaVerou committed:
Fix bugs

LeaVerou committed:
Various fixes

LeaVerou committed:
[Tab] Avoid sprouting attributes in the constructor

LeaVerou committed:
Drop `updateContrastTables()`, update contrast tables reactively

LeaVerou committed:
Make code examples reactive too

LeaVerou committed:
Code block improvements

- Add ids, use ids to link copy button. No need for client-side script or updating the copy button manually for dynamic code snippets.
- Add button to link to code block
- Slight refactor on copy plugin to use the 11ty API properly

LeaVerou committed:
Remove code setting copyButton.value

LeaVerou committed:
Fixes & refactor sidebar code

LeaVerou committed:
Nicer UI

LeaVerou committed:
Reset all tweaks button

LeaVerou committed:
Palette code does not need to include actual tweaks!

LeaVerou committed:
Add maxChromaTint properties to `this.colors`

LeaVerou committed:
Human readable description of tweaks

LeaVerou committed:
Add missing utils

LeaVerou committed:
Nicer loading

LeaVerou committed:
Don't tweak colors in place!

LeaVerou committed:
Tweaks as removable tags

LeaVerou committed:
Gray tweaks prototype

LeaVerou committed:
Merge branch 'next' into gray-tweaks

LeaVerou committed:
Iterate on gray tweaking code

LeaVerou committed:
Use different ranges for different hues

lindsaym-fa commented on docs/docs/palettes/tweak.js:
```diff
@@ -4,8 +4,24 @@ import { createApp, nextTick } from 'https://unpkg.com/vue@3/dist/vue.esm-browse
+  yellow: 0.25,
+  green: 0.2,
+  cyan: 0.3,
+  blue: 0.3,
```
```suggestion
  blue: 0.35,
```

lindsaym-fa commented on docs/docs/palettes/tweak.js:
```diff
@@ -4,8 +4,24 @@ import { createApp, nextTick } from 'https://unpkg.com/vue@3/dist/vue.esm-browse
+  red: 0.2,
+  orange: 0.2,
+  yellow: 0.25,
+  green: 0.2,
```
```suggestion
  green: 0.25,
```

lindsaym-fa reviewed (commented):
Looking good! Just left suggestions to bump up the max gray chroma from `green` and `blue` undertones.

It looks like we may have broken the "Overall colorfulness" slider? It doesn't seem to have any affect in my testing.

LeaVerou committed:
Update docs/docs/palettes/tweak.js

Co-authored-by: Lindsay M <126139086+lindsaym-fa@users.noreply.github.com>

LeaVerou committed:
Update docs/docs/palettes/tweak.js

Co-authored-by: Lindsay M <126139086+lindsaym-fa@users.noreply.github.com>

LeaVerou committed:
Add Pro badge on all tweaked palettes, rel #764

LeaVerou committed:
Fix bug with chroma scale

LeaVerou committed:
Fix error when navigating away from palette

LeaVerou committed:
Add Pro badge to Pro & remixed themes (fixes #764)

LeaVerou commented:
Just fixed the overall colorfulness slider. It was the stupidest thing ever, just a good ol' typo 🤦🏽‍♀️ : 
<img width="398" alt="image" src="https://github.com/user-attachments/assets/fff93c70-f5f3-4158-9fa4-6f3995c805f7" />

I also added a Pro badge to every tweaked color palette, every pro theme (yup, there wasn't one on the theme page!), and every remixed theme (fixes #764)

lindsaym-fa reviewed (commented):
A couple of remaining issues (that seem like they're probably related), then I think we have it!

 ---

1) When tweaks are reset, the changes to gray persist.

https://github.com/user-attachments/assets/eef0be31-835f-4c05-a155-2497b64aa11d

---

2) Changes to gray are not autosaved. E.g., when I edit grays on a saved palette, navigate away, and navigate back, the changes don't persist.

https://github.com/user-attachments/assets/8f54a9c8-791d-4d82-bde5-92c0d7f5e36d

LeaVerou committed:
Remove dead code

LeaVerou committed:
Make sure reset rests gray tweaks too

LeaVerou committed:
Fix error

LeaVerou committed:
Move data to data file

LeaVerou committed:
Drop warmer/cooler, use actual hue

LeaVerou committed:
Update demo.njk

LeaVerou committed:
Fix saving gray tweaks

LeaVerou committed:
Reset gray tweaks (+ refactor around reseting)

LeaVerou committed:
Prevent error on second render

LeaVerou committed:
Fix various bugs around palette saving & current links

LeaVerou commented:
Oof, that took a while! I just pushed a fix that fixes a bunch of bugs around saving, current links, permalinks etc, including (but not limited to) the two you found!

lindsaym-fa reviewed (commented):
One more for you. 🥲

When I edit a saved palette, either by removing a tweak or tweaking something new, the selected link in the sidebar switches to the original palette, and the new modification doesn't take. E.g.:

https://github.com/user-attachments/assets/2354c18e-3cc6-4844-a512-883f81051641

LeaVerou committed:
`removeTweak()` -> `reset()`

LeaVerou committed:
More pink -> Pinker

LeaVerou committed:
Fix bug

LeaVerou commented:
🫠 
So sorry for the multiple iterations! I think I fixed it now.

lindsaym-fa commented:
It looks like the "Reset" button no longer works (though I'll take a closer look — I may just need to clear my cache). We could simply remove it for this iteration and re-add it later, if it's not worth the energy to troubleshoot now.

LeaVerou committed:
Fix Reset button

LeaVerou commented:
> It looks like the "Reset" button no longer works (though I'll take a closer look — I may just need to clear my cache). We could simply remove it for this iteration and re-add it later, if it's not worth the energy to troubleshoot now.

It wasn't your cache! Just fixed it, thankfully it was a simple fix.

LeaVerou committed:
Merge branch 'next' into gray-tweaks

lindsaym-fa reviewed (approved):
All looks good! 🎉
